src/main/java/org/elasticsearch/indices/recovery/RecoveryTarget.java

```diff
@@ -157,13 +158,24 @@ private void doRecovery(final RecoveryStatus recoveryStatus) {
 
         logger.trace("collecting local files for {}", recoveryStatus);
         final Map<String, StoreFileMetaData> existingFiles;
-        try {
-            existingFiles = recoveryStatus.store().getMetadataOrEmpty().asMap();
-        } catch (Exception e) {
-            logger.debug("error while listing local files, recovery as if there are none", e);
-            onGoingRecoveries.failRecovery(recoveryStatus.recoveryId(),
-                    new RecoveryFailedException(recoveryStatus.state(), "failed to list local files", e), true);
-            return;
+        final Version sourceNodeVersion = recoveryStatus.sourceNode().version();
+        if (sourceNodeVersion.before(Version.V_1_3_2) && recoverySettings.compress()) { // don't recover from pre 1.3.2 if compression is on?
+            throw new ElasticsearchIllegalStateException("Can't recovery from node "
+                    + recoveryStatus.sourceNode() + " with [" + RecoverySettings.INDICES_RECOVERY_COMPRESS
+                    + " : true] due to compression bugs -  see issue #7210 for details" );
+        }
+        if (sourceNodeVersion.onOrAfter(Version.V_1_4_0)) {
+            try {
+                existingFiles = recoveryStatus.store().getMetadataOrEmpty().asMap();
+            } catch (Exception e) {
+                logger.debug("error while listing local files, recovery as if there are none", e);
```

Comment:
Remove the ", recovery as if there are none"?  Because we are failing the recovery instead right?